File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/WorkspaceSearchCriteria.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * List of Workspace objects
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=WorkspaceSearchCriteria.Builder.class)

public final class WorkspaceSearchCriteria  {
    @Deprecated
    @java.beans.ConstructorProperties({"path", "query", "limit", "offset", "aggregations", "filter", "sort"})
    public WorkspaceSearchCriteria(String path, String query, Integer limit, Integer offset, java.util.List<AggregationEnum> aggregations, FilterTypes filter, java.util.List<SortDetails> sort) {
        super();
        this.path = path;
        this.query = query;
        this.limit = limit;
        this.offset = offset;
        this.aggregations = aggregations;
        this.filter = filter;
        this.sort = sort;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The path prefix Example: /Shared, /Shared/Folder1, etc.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("path")
private String path;

        /**
         * The path prefix Example: /Shared, /Shared/Folder1, etc.
* 
         * @param path the value to set
         * @return this builder
         **/
        

public Builder path(String path) {
    this.path = path;
    return this;
}
            /**
     * Search query string
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("query")
private String query;

        /**
         * Search query string
         * @param query the value to set
         * @return this builder
         **/
        

public Builder query(String query) {
    this.query = query;
    return this;
}
            /**
     * The maximum number of items to return.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("limit")
private Integer limit;

        /**
         * The maximum number of items to return.
         * @param limit the value to set
         * @return this builder
         **/
        

public Builder limit(Integer limit) {
    this.limit = limit;
    return this;
}
            /**
     * The doc number from which it needs to be return.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("offset")
private Integer offset;

        /**
         * The doc number from which it needs to be return.
         * @param offset the value to set
         * @return this builder
         **/
        

public Builder offset(Integer offset) {
    this.offset = offset;
    return this;
}
            /**
     * The provide the field name on which we need aggregation.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("aggregations")
private java.util.List<AggregationEnum> aggregations;

        /**
         * The provide the field name on which we need aggregation.
         * @param aggregations the value to set
         * @return this builder
         **/
        

public Builder aggregations(java.util.List<AggregationEnum> aggregations) {
    this.aggregations = aggregations;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("filter")
private FilterTypes filter;



public Builder filter(FilterTypes filter) {
    this.filter = filter;
    return this;
}
            /**
     * Array of objects having details about sort field and order.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("sort")
private java.util.List<SortDetails> sort;

        /**
         * Array of objects having details about sort field and order.
         * @param sort the value to set
         * @return this builder
         **/
        

public Builder sort(java.util.List<SortDetails> sort) {
    this.sort = sort;
    return this;
}


        public WorkspaceSearchCriteria build() {
            WorkspaceSearchCriteria model = new WorkspaceSearchCriteria(this.path
                , this.query
                , this.limit
                , this.offset
                , this.aggregations
                , this.filter
                , this.sort);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(WorkspaceSearchCriteria model) {
                this.path(model.getPath());
    this.query(model.getQuery());
    this.limit(model.getLimit());
    this.offset(model.getOffset());
    this.aggregations(model.getAggregations());
    this.filter(model.getFilter());
    this.sort(model.getSort());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The path prefix Example: /Shared, /Shared/Folder1, etc.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("path")
    private final String path;

        /**
     * The path prefix Example: /Shared, /Shared/Folder1, etc.
* 
     * @return the value
     **/
    
    public String getPath() {
        return path;
    }


        /**
     * Search query string
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("query")
    private final String query;

        /**
     * Search query string
     * @return the value
     **/
    
    public String getQuery() {
        return query;
    }


        /**
     * The maximum number of items to return.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("limit")
    private final Integer limit;

        /**
     * The maximum number of items to return.
     * @return the value
     **/
    
    public Integer getLimit() {
        return limit;
    }


        /**
     * The doc number from which it needs to be return.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("offset")
    private final Integer offset;

        /**
     * The doc number from which it needs to be return.
     * @return the value
     **/
    
    public Integer getOffset() {
        return offset;
    }


        /**
     * The provide the field name on which we need aggregation.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("aggregations")
    private final java.util.List<AggregationEnum> aggregations;

        /**
     * The provide the field name on which we need aggregation.
     * @return the value
     **/
    
    public java.util.List<AggregationEnum> getAggregations() {
        return aggregations;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("filter")
    private final FilterTypes filter;

    
    public FilterTypes getFilter() {
        return filter;
    }


        /**
     * Array of objects having details about sort field and order.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("sort")
    private final java.util.List<SortDetails> sort;

        /**
     * Array of objects having details about sort field and order.
     * @return the value
     **/
    
    public java.util.List<SortDetails> getSort() {
        return sort;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("WorkspaceSearchCriteria(");
        sb.append("path=").append(String.valueOf(this.path));
        sb.append(", query=").append(String.valueOf(this.query));
        sb.append(", limit=").append(String.valueOf(this.limit));
        sb.append(", offset=").append(String.valueOf(this.offset));
        sb.append(", aggregations=").append(String.valueOf(this.aggregations));
        sb.append(", filter=").append(String.valueOf(this.filter));
        sb.append(", sort=").append(String.valueOf(this.sort));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof WorkspaceSearchCriteria)) {
            return false;
        }

        WorkspaceSearchCriteria other = (WorkspaceSearchCriteria) o;
        return java.util.Objects.equals(this.path, other.path) &&
            java.util.Objects.equals(this.query, other.query) &&
            java.util.Objects.equals(this.limit, other.limit) &&
            java.util.Objects.equals(this.offset, other.offset) &&
            java.util.Objects.equals(this.aggregations, other.aggregations) &&
            java.util.Objects.equals(this.filter, other.filter) &&
            java.util.Objects.equals(this.sort, other.sort);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.path == null ? 43 : this.path.hashCode());
        result = (result * PRIME) + (this.query == null ? 43 : this.query.hashCode());
        result = (result * PRIME) + (this.limit == null ? 43 : this.limit.hashCode());
        result = (result * PRIME) + (this.offset == null ? 43 : this.offset.hashCode());
        result = (result * PRIME) + (this.aggregations == null ? 43 : this.aggregations.hashCode());
        result = (result * PRIME) + (this.filter == null ? 43 : this.filter.hashCode());
        result = (result * PRIME) + (this.sort == null ? 43 : this.sort.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/workspace_search_criteria.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class WorkspaceSearchCriteria(object):
    """
    List of Workspace objects
    """

    def __init__(self, **kwargs):
        """
        Initializes a new WorkspaceSearchCriteria object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param path:
            The value to assign to the path property of this WorkspaceSearchCriteria.
        :type path: str

        :param query:
            The value to assign to the query property of this WorkspaceSearchCriteria.
        :type query: str

        :param limit:
            The value to assign to the limit property of this WorkspaceSearchCriteria.
        :type limit: int

        :param offset:
            The value to assign to the offset property of this WorkspaceSearchCriteria.
        :type offset: int

        :param aggregations:
            The value to assign to the aggregations property of this WorkspaceSearchCriteria.
        :type aggregations: list[oci.aidataplatform_dp.models.AggregationEnum]

        :param filter:
            The value to assign to the filter property of this WorkspaceSearchCriteria.
        :type filter: oci.aidataplatform_dp.models.FilterTypes

        :param sort:
            The value to assign to the sort property of this WorkspaceSearchCriteria.
        :type sort: list[oci.aidataplatform_dp.models.SortDetails]

        """
        self.swagger_types = {
            'path': 'str',
            'query': 'str',
            'limit': 'int',
            'offset': 'int',
            'aggregations': 'list[AggregationEnum]',
            'filter': 'FilterTypes',
            'sort': 'list[SortDetails]'
        }

        self.attribute_map = {
            'path': 'path',
            'query': 'query',
            'limit': 'limit',
            'offset': 'offset',
            'aggregations': 'aggregations',
            'filter': 'filter',
            'sort': 'sort'
        }

        self._path = None
        self._query = None
        self._limit = None
        self._offset = None
        self._aggregations = None
        self._filter = None
        self._sort = None

    @property
    def path(self):
        """
        Gets the path of this WorkspaceSearchCriteria.
        The path prefix Example: /Shared, /Shared/Folder1, etc.


        :return: The path of this WorkspaceSearchCriteria.
        :rtype: str
        """
        return self._path

    @path.setter
    def path(self, path):
        """
        Sets the path of this WorkspaceSearchCriteria.
        The path prefix Example: /Shared, /Shared/Folder1, etc.


        :param path: The path of this WorkspaceSearchCriteria.
        :type: str
        """
        self._path = path

    @property
    def query(self):
        """
        Gets the query of this WorkspaceSearchCriteria.
        Search query string


        :return: The query of this WorkspaceSearchCriteria.
        :rtype: str
        """
        return self._query

    @query.setter
    def query(self, query):
        """
        Sets the query of this WorkspaceSearchCriteria.
        Search query string


        :param query: The query of this WorkspaceSearchCriteria.
        :type: str
        """
        self._query = query

    @property
    def limit(self):
        """
        Gets the limit of this WorkspaceSearchCriteria.
        The maximum number of items to return.


        :return: The limit of this WorkspaceSearchCriteria.
        :rtype: int
        """
        return self._limit

    @limit.setter
    def limit(self, limit):
        """
        Sets the limit of this WorkspaceSearchCriteria.
        The maximum number of items to return.


        :param limit: The limit of this WorkspaceSearchCriteria.
        :type: int
        """
        self._limit = limit

    @property
    def offset(self):
        """
        Gets the offset of this WorkspaceSearchCriteria.
        The doc number from which it needs to be return.


        :return: The offset of this WorkspaceSearchCriteria.
        :rtype: int
        """
        return self._offset

    @offset.setter
    def offset(self, offset):
        """
        Sets the offset of this WorkspaceSearchCriteria.
        The doc number from which it needs to be return.


        :param offset: The offset of this WorkspaceSearchCriteria.
        :type: int
        """
        self._offset = offset

    @property
    def aggregations(self):
        """
        Gets the aggregations of this WorkspaceSearchCriteria.
        The provide the field name on which we need aggregation.


        :return: The aggregations of this WorkspaceSearchCriteria.
        :rtype: list[oci.aidataplatform_dp.models.AggregationEnum]
        """
        return self._aggregations

    @aggregations.setter
    def aggregations(self, aggregations):
        """
        Sets the aggregations of this WorkspaceSearchCriteria.
        The provide the field name on which we need aggregation.


        :param aggregations: The aggregations of this WorkspaceSearchCriteria.
        :type: list[oci.aidataplatform_dp.models.AggregationEnum]
        """
        self._aggregations = aggregations

    @property
    def filter(self):
        """
        Gets the filter of this WorkspaceSearchCriteria.

        :return: The filter of this WorkspaceSearchCriteria.
        :rtype: oci.aidataplatform_dp.models.FilterTypes
        """
        return self._filter

    @filter.setter
    def filter(self, filter):
        """
        Sets the filter of this WorkspaceSearchCriteria.

        :param filter: The filter of this WorkspaceSearchCriteria.
        :type: oci.aidataplatform_dp.models.FilterTypes
        """
        self._filter = filter

    @property
    def sort(self):
        """
        Gets the sort of this WorkspaceSearchCriteria.
        Array of objects having details about sort field and order.


        :return: The sort of this WorkspaceSearchCriteria.
        :rtype: list[oci.aidataplatform_dp.models.SortDetails]
        """
        return self._sort

    @sort.setter
    def sort(self, sort):
        """
        Sets the sort of this WorkspaceSearchCriteria.
        Array of objects having details about sort field and order.


        :param sort: The sort of this WorkspaceSearchCriteria.
        :type: list[oci.aidataplatform_dp.models.SortDetails]
        """
        self._sort = sort

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
